common/metrics/src/lib.rs

```diff
@@ -216,3 +216,27 @@ macro_rules! monitor {
         result
     }};
 }
+
+/// Helper function to increment a counter easily
+#[macro_export]
+macro_rules! inc {
+    ($counter:expr) => {
+        $counter.inc()
+    };
+    ($counter:expr, $value:expr) => {
+        $counter.with_label_values(&[$value]).inc()
+    };
+    ($counter:expr, $value1:expr, $value2:expr) => {
+        $counter.with_label_values(&[$value1, $value2]).inc()
+    };
+    ($counter:expr, $value1:expr, $value2:expr, $value3:expr) => {
+        $counter
+            .with_label_values(&[$value1, $value2, $value3])
+            .inc()
+    };
+    ($counter:expr, $value1:expr, $value2:expr, $value3:expr, $value4:expr) => {
+        $counter
+            .with_label_values(&[$value1, $value2, $value3, $value4])
+            .inc()
+    };
+}
```

Comment:
Could we instead explore a typed interface into metrics, see: https://github.com/diem/diem/blob/master/state-sync/src/counters.rs#L21-L25

this prevents accidental metric breakage

Reply:
Most of them being used are `&'static str` spewed all over the code base.  I was trying to add less boilerplate code, so I'll probably just remove this commit for now :P